### ISSUE TITLE
0.3.14 breaks addOrderBy

### ISSUE THREAD

**Issue by paoesco** [typeorm/typeorm#9960]
### Issue description

0.3.14 introduced a change where the alias is not mapped correctly to SQL equivalent 

### Expected Behavior

The following code:
```
query = query.addOrderBy("verification.createdAt", "DESC", "NULLS LAST");
```
should produce the following SQL:
```
ORDER BY "verification"."created_at" DESC NULLS LAST LIMIT 10
```

### Actual Behavior

The following code:
```
query = query.addOrderBy("verification.createdAt", "DESC", "NULLS LAST");
```
is translated to SQL:
```
ORDER BY "verification_created_at" [object Object] LIMIT 10
```

### Steps to reproduce

Create a query with `createQueryBuilder` and add `addOrderBy`.

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |  Mac OS     |
| Node.js version     | 18.15.0  |
| Typescript version  |  4.8.4  |
| TypeORM version     | 0.3.14 |

### Additional Context

I believe the issue has been introduced by the following PR: https://github.com/typeorm/typeorm/pull/9914/files

### Relevant Database Driver(s)

- [ ] aurora-mysql
- [ ] aurora-postgres
- [ ] better-sqlite3
- [ ] cockroachdb
- [ ] cordova
- [ ] expo
- [ ] mongodb
- [ ] mysql
- [ ] nativescript
- [ ] oracle
- [X] postgres
- [ ] react-native
- [ ] sap
- [ ] spanner
- [ ] sqlite
- [ ] sqlite-abstract
- [ ] sqljs
- [ ] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

Yes, I have the time, but I don't know how to start. I would need guidance.

**Comment by pleerock** [comment]
Added test in #9961 

It does work for me on my local machine, let's see if CI tests pass.

Meanwhile, please provide a minimal reproduction on the issue.

**Comment by pleerock** [comment]
@paoesco please review the PR and let me know what code changes I need to add to make your issues reproducible. As you can see tests are passing

**Comment by pleerock** [comment]
Confirmed. Will be fixed by #9961 and merged very soon.

**Comment by paoesco** [comment]
Quite impressed by how fast it got resolved.
Thanks team!
